keyboards/kprepublic/bm60hsrgb_ec/rev2/keymaps/default/keymap.c

```diff
@@ -0,0 +1,62 @@
+/* Copyright 2021 bdtc123
+ * Copyright 2022 peepeetee
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_60_ansi_arrow(
+        KC_GESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,
+        KC_TAB,      KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+        KC_CAPS,       KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_MUTE, KC_ENT,
+        KC_LSFT,          KC_Z,   KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_RSFT,        KC_UP,   KC_SLSH,
+        KC_LCTL,   KC_LGUI,   KC_LALT,                       KC_SPC,                              KC_RALT, MO(1),   KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+    [1] = LAYOUT_60_ansi_arrow(
+        _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  RESET,
+        _______,     RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, _______, _______, _______, KC_DEL,
+        _______,            _______, _______,  _______,  _______,  _______,  _______,  _______,  _______, _______, _______, _______, _______, _______,
+        _______,            _______, _______, _______, _______, _______, NK_TOGG, _______, _______, _______, _______,        KC_PGUP, KC_INS,
+        _______,   _______,   _______,                      _______,                              _______, _______, KC_HOME, KC_PGDN, KC_END
+    )
+
+};
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (IS_LAYER_ON(1)){
+    if (clockwise) {
+        tap_code(KC_VOLU);
+    } else {
+        tap_code(KC_VOLD);
+    }
+  } else {
+      // ctrl
+      if (get_mods() & MOD_BIT(KC_LCTRL)) {
+        if (clockwise) {
+          tap_code(KC_WH_R);
+        } else {
+          tap_code(KC_WH_L);
+        }
+      } else {
+        if (clockwise) {
+          tap_code(KC_WH_D);
+        } else {
+          tap_code(KC_WH_U);
+        }
+      }
+  }
+
+  return false;
+}
```

Comment:
```suggestion
```

Reply:
Would this change disable the encoder's functionality? Is it best practice to leave that functionality undefined and in the default keymap so the user defines that functionality? Please advise.